firebase/test/core/data/schemas/memo.spec.ts

```diff
@@ -2,153 +2,67 @@ import Ajv2020 from "ajv/dist/2020";
-  });
-
-  it("should throw when question id is not present", () => {
-    const rawMemo = _newRawMemo();
-
-    delete rawMemo.id;
-
-    throws(() => validator.validateObject("memo", rawMemo), SerializationError);
-  });
-
-  describe("Question - ", () => {
-    it("should throw when question is not present", () => {
-      const rawMemo = _newRawMemo();
-
-      delete rawMemo.question;
-
-      throws(() => validator.validateObject("memo", rawMemo), SerializationError);
-    });
-
-    it("should throw when question content is not present", () => {
-      const rawMemo = _newRawMemo();
-
-      delete rawMemo.question[0].insert;
-
-      throws(() => validator.validateObject("memo", rawMemo), SerializationError);
-    });
-
-    it("should throw when question is empty", () => {
-      const rawMemo = _newRawMemo();
-
-      rawMemo.question = [];
-
-      throws(() => validator.validateObject("memo", rawMemo), SerializationError);
+  const allowedAttributesProperties = ["bold", "italic", "underline", "code-block"];
+
+  describe("Root Properties - ", () => {
+    const properties: ValidationProperties = {
+      required: ["id", "question", "answer"],
+      array: ["question", "answer"],
+      uniqueItems: new Map<string, any[]>([
+        ["question", [newRawMemoContent(), newRawMemoContent()]],
+        ["answer", [newRawMemoContent(), newRawMemoContent()]],
+      ]),
+      incorrectTypes: new Map<string, any>([
+        ["id", true],
+        ["question", "string"],
+        ["answer", "string"],
+      ]),
+    };
+
+    const validator = new SchemaValidatorBuilder({
+      schema: "memo",
+      entityConstructor: newRawMemo,
+      properties: properties,
     });
 
-    it("should throw when question has incorrect type", () => {
-      const rawMemo = _newRawMemo();
-
-      rawMemo.question = true;
-
-      throws(() => validator.validateObject("memo", rawMemo), SerializationError);
-    });
-  });
-
-  describe("Answer - ", () => {
-    it("should throw when answer is not present", () => {
-      const rawMemo = _newRawMemo();
-
-      delete rawMemo.answer;
-
-      throws(() => validator.validateObject("memo", rawMemo), SerializationError);
-    });
-
-    it("should throw when answer content is not present", () => {
-      const rawMemo = _newRawMemo();
-
-      delete rawMemo.answer[0].insert;
-
-      throws(() => validator.validateObject("memo", rawMemo), SerializationError);
-    });
-
-    it("should throw when answer is empty", () => {
-      const rawMemo = _newRawMemo();
-
-      delete rawMemo.answer;
-
-      throws(() => validator.validateObject("memo", rawMemo), SerializationError);
-    });
-
-    it("should throw when answer has incorrect type", () => {
-      const rawMemo = _newRawMemo();
-
-      rawMemo.answer = true;
-
-      throws(() => validator.validateObject("memo", rawMemo), SerializationError);
-    });
+    validator.validate();
   });
 
   describe("Attributes - ", () => {
```

Comment:
This `Attributes` could have been separated in a separate schema to also use the validator, but I think that it's not worth it to be honest, so I left this specific tests here